wildfly-users-jpa/src/main/java/model/BaseEntity.java

```diff
@@ -1,12 +1,41 @@
 package model;
 
 import java.io.Serializable;
+import java.util.UUID;
 
-public abstract class BaseEntity implements Serializable{
+import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
+import javax.persistence.PrePersist;
 
+@MappedSuperclass
+public abstract class BaseEntity implements Serializable{
 	/**
 	 * 
 	 */
-	private static final long serialVersionUID = 456302430148243749L;
+	private static final long serialVersionUID = 3333408811149331428L;
+	
+	@Id
+	private String uuid;
+
+	public String getUuid() {
+		if (this.uuid == null) {

```

Comment:
csak a onPrePersist-ben lehet h nem null es akkor is legeneralna, vagy eleg ha a getUuid-t hivom ott is?